shared/zodSchemas.ts

```diff
@@ -0,0 +1,52 @@
+import { z } from 'zod';
+import { emailRegex, passwordRegex } from './regexPatterns';
+import { Prisma } from '@prisma/client';
+
+export const giftSchema = z.object({
+  gift: z.string().min(1),
+  receiver: z.string().min(1),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+  uuid: z.string(),
+});
+
+export const createGiftSchema = z.object({
+  gift: z.string().min(1),
+  receiver: z.string().min(1),
+});
+
+// CHECK THIS
+// onko alla oleva rivi tarpeellinen?
+export const updateGiftSchema = z.object({
+  gift: z.string().min(1),
+  receiver: z.string().min(1),
+});
```

Comment:
Tähän olin merkannut, että onko tarpeellinen. Tämä rivi on kyllä käytössä, mutta ajatuksena oli se että on täysin sama tällä hetkellä kuin ylempänä oleva `createGiftSchema`

Reply:
Sanoisin että tähän on kaksi hyvää tapaa:
1. Käyttää vain yhtää skeemaa ja nimeää `createGiftSchema` -> `createOrUpdateGiftSchema`
2. Käyttää kahta muuttujaa, mutta määrittelee kentät vain kerran, eli näin:
Kumpikin yhtä hyvä tapa, eli valitse vapaasti.
```suggestion
export const updateGiftSchema = createGiftSchema;
```

Reply:
Tapa 2. kuulostaa hyvältä. Jos jossain kohtia tulee tarve tehdä muutoksia, on se helpompi toteuttaa, 2. tavalla. Kiitos! :+1: